### DISCUSSION TITLE
"saving session...copying shared history....saving history...."

### DISCUSSION THREAD

**Discussion by nfaltir** [pyinstaller/pyinstaller#6225] [Help]
If you compile a simple python script that generates a random password, does `pyinstaller ` save the output after you run the script?

I am getting this message on the terminal after running my python script:

<code>
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
</code>
 <br />

<br />

I am assuming that the above message is indicating that the script has finished running, I am just curious whether it saves the output or not. 

Any help is super appreciated, thanks 👌
machine: mac

**Comment by rokm** [comment]
> If you compile a simple python script that generates a random password, does `pyinstaller ` save the output after you run the script?

PyInstaller just runs your frozen script, so whatever is stored or not is up to your code and the 3rd party packages you use. Or whoever/whatever calls the frozen program, if they happen to capture or redirect stdout and stderr.

> I am getting this message on the terminal after running my python script:
> ` Saving session... ...copying shared history... ...saving history...truncating history files... ...completed. `

Those are [messages from your macOS shell/terminal](https://superuser.com/questions/975678/strange-output-from-terminal-exit-command-is-this-a-virus/975731), and have nothing to do with PyInstaller. And they are probably referring to saving/syncing command history, rather than saving the command's stdout/stderr..

**Reply by nfaltir** [reply]
ah, I see,

thank you 🎉

**Reply by edrozenberg** [reply]
I put `SHELL_SESSIONS_DISABLE=1` in my `.zprofile` (I'm using zsh on Sequoia) to get rid of the annoying history cr*p delays when existing terminal tabs/windows. Downside is no history but I don't need it.
